app/controllers/users/two_factor_verification_controller.rb

```diff
@@ -0,0 +1,57 @@
+module Users
+  class TwoFactorVerificationController < ApplicationController
+    before_action :check_direct_otp_expiry, only: [:show, :create]
+
+
+    def show
+      unless current_user.totp_enabled?
+        current_user.create_direct_otp
+        r = SMSService.new.send_two_factor_confirmation_code!(current_user)
+
+        # TODO: make this async and handle service failure
+        unless r.code == 202
+          raise 'There was an error sending the SMS'
```

Comment:
As we discussed on slack, fire the SMS request off in a new thread and ignore it if it fails to send, but indicate to the user that they can try again if they do not receive an SMS.